lib/active_record/acts_as/class_methods.rb

```diff
@@ -16,7 +16,7 @@ def validators_on(*args)
       end
 
       def actables
-        acting_as_model.where(actable_id: select(:id))
+        acting_as_model.where(actable_id: select(:id), actable_type: model_name.name)
```

Comment:
Can you byebug here, run a random test, and show me the result of `acting_as_model.where(actable_id: select(:id))` and `acting_as_model.where(actable_id: select(:id), actable_type: model_name.name)`?

I want to see the difference

Reply:
Yes sure, I'm using the test inside `acts_as_spec.rb`, printing the result of both with and without including `actable_type: model_name.name`, and here are the results:

Without `actable_type: model_name.name`
`[#<Product id: 18, name: "red pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 01:34:34.894579000 +0000", updated_at: "2024-06-03 01:34:34.894579000 +0000", actable_type: "Pen", actable_id: 19>, #<Product id: 19, name: "blue pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 01:34:34.895272000 +0000", updated_at: "2024-06-03 01:34:34.895272000 +0000", actable_type: "Pen", actable_id: 20>]`

With `actable_type: model_name.name`
`[#<Product id: 18, name: "red pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 01:34:34.894579000 +0000", updated_at: "2024-06-03 01:34:34.894579000 +0000", actable_type: "Pen", actable_id: 19>, #<Product id: 19, name: "blue pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 01:34:34.895272000 +0000", updated_at: "2024-06-03 01:34:34.895272000 +0000", actable_type: "Pen", actable_id: 20>]`

Actually if you see here, the results are exactly the same. However, I remembered in our DB there are several entities inside `Course::Assessment::Question` that shares the same `actable_id` but belongs to different `actable_type`, therefore I feel the need to add the specification of `actable_type` also here, following what chaadow did

Reply:
What if there’s 2 entities with same actable_id and different actable_type?
Can show me the print result diff?

Reply:
Sure do.. I have added 20 entities of Eraser (which is a new model I've defined, also acts_as product), having `actable_id` from 1 to 20 and `actable_type` to be `Eraser`, here is the result

without actable_type 
`[#<Product id: 18, name: "red pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 02:18:51.118111000 +0000", updated_at: "2024-06-03 02:18:51.118111000 +0000", actable_type: "Pen", actable_id: 19>, #<Product id: 19, name: "blue pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 02:18:51.118806000 +0000", updated_at: "2024-06-03 02:18:51.118806000 +0000", actable_type: "Pen", actable_id: 20>, #<Product id: 39, name: "eraser", price: 1.2, store_id: nil, settings: {}, created_at: "2024-06-03 02:18:51.133374000 +0000", updated_at: "2024-06-03 02:18:51.133374000 +0000", actable_type: "Eraser", actable_id: 19>, #<Product id: 40, name: "eraser", price: 1.2, store_id: nil, settings: {}, created_at: "2024-06-03 02:18:51.134054000 +0000", updated_at: "2024-06-03 02:18:51.134054000 +0000", actable_type: "Eraser", actable_id: 20>]`

with actable_type 
`[#<Product id: 18, name: "red pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 02:18:51.118111000 +0000", updated_at: "2024-06-03 02:18:51.118111000 +0000", actable_type: "Pen", actable_id: 19>, #<Product id: 19, name: "blue pen", price: 0.8, store_id: nil, settings: {}, created_at: "2024-06-03 02:18:51.118806000 +0000", updated_at: "2024-06-03 02:18:51.118806000 +0000", actable_type: "Pen", actable_id: 20>]`

Reply:
Im curious why this didnt happen before, or perhaps it has happened previously? If so, why it didnt affect our app negatively?. Anyway, can add additional tests for this case.

Reply:
after checking our codebase, it's revealed that this function has never been called (we have never called `actables` in our codebase). Only `actable` is used over there.
However, I think adding this line should be crucial, so that next time we actually use this function, it returns the correct answer.

Yes, I have added the instance of multiple Erasers into the testcase, I think that should be enough.

Reply:
Ok. So actable is not affected right? 

The lesson here is that you should be able to come up with the questions I asked by yourself.

Reply:
Yes, it's not affected by this.